CB-14796 Remove RBAC entitlement and remove legacy authorization code

horadla23 opened the pull request:
See detailed description in the commit message.

bergerdenes reviewed (approved):
Minor nits in readme formatting

horadla23 commented on authorization-common/README.md:
```diff
@@ -254,13 +252,12 @@ You can support authorization on new resources, and you can specify the rights a
 2. define new rights by extending `AuthorizationResourceAction` enum with new (String right, AuthorizationResourceType type) values (the new right should be defined in UMS previously),
-3. to support legacy authorization add a mapping from the new right to the legacy right in the `legacyRights.json`,
-4. implement `ResourceBasedCrnProvider`'s methods to support different scenarios on API level,
+3implement `ResourceBasedCrnProvider`'s methods to support different scenarios on API level,
```
good catch, thx!

lnardai commented:
Please disable legacy real-ums-test before merge.

horadla23 commented:
> Please disable legacy real-ums-test before merge.

I have removed references for legacy authz test suite in jenkins jobs and removed the related suite file and test in this PR.
